Gems/ROS2/Code/Source/RobotImporter/FixURDF/FixURDF.cpp

```diff
@@ -16,53 +16,94 @@ namespace ROS2::Utils
-            inertia->append_attribute(urdf->document()->allocate_attribute("iyy", "1."));
-            inertia->append_attribute(urdf->document()->allocate_attribute("iyz", "0."));
+            auto inertia = inertial->first_node("inertia");
+            if (!inertia)
+            {
+                inertia = urdf->document()->allocate_node(node_element, "inertia");
 
-            inertia->append_attribute(urdf->document()->allocate_attribute("izx", "0."));
-            inertia->append_attribute(urdf->document()->allocate_attribute("izy", "0."));
-            inertia->append_attribute(urdf->document()->allocate_attribute("izz", "1."));
+                inertia->append_attribute(urdf->document()->allocate_attribute("ixx", "1."));
+                inertia->append_attribute(urdf->document()->allocate_attribute("ixy", "0."));
+                inertia->append_attribute(urdf->document()->allocate_attribute("ixz", "0."));
 
-            inertial->append_node(inertia);
+                inertia->append_attribute(urdf->document()->allocate_attribute("iyx", "0."));
+                inertia->append_attribute(urdf->document()->allocate_attribute("iyy", "1."));
+                inertia->append_attribute(urdf->document()->allocate_attribute("iyz", "0."));
 
-            xml_node<>* origin = urdf->document()->allocate_node(node_element, "origin");
-            origin->append_attribute(urdf->document()->allocate_attribute("xyz", "0. 0. 0."));
-            inertial->append_node(origin);
+                inertia->append_attribute(urdf->document()->allocate_attribute("izx", "0."));
+                inertia->append_attribute(urdf->document()->allocate_attribute("izy", "0."));
+                inertia->append_attribute(urdf->document()->allocate_attribute("izz", "1."));
 
-            auto* name = link->first_attribute("name");
-            if (name)
+                inertial->append_node(inertia);
+                if (!lacks_inertial)
+                {
+                    modifiedLinksTags[name].push_back("inertial->inertia");
+                }
+            }
+
+            auto origin = inertial->first_node("origin");
+            if (!origin)
+            {
+                origin = urdf->document()->allocate_node(node_element, "origin");
+                origin->append_attribute(urdf->document()->allocate_attribute("xyz", "0. 0. 0."));
+                inertial->append_node(origin);
+
+                if (!lacks_inertial)
+                {
+                    modifiedLinksTags[name].push_back("inertial->origin");
+                }
+            }
+        }
+        AZStd::vector<AZStd::string> modifiedLinks;
+
+        for (auto [link, tags] : modifiedLinksTags)
+        {
+            AZStd::string modified = link + ":";
+
+            for (auto tag : tags)
             {
-                modifiedLinks.push_back(name->value());
+                modified += " " + tag + ", ";
```

Comment:
I think that needs to be addressed in a more systematic way. Currently, it is even more confusing. Please revert this logic of return value here. 
Let it fix correctly in next PR, ok?

Reply:
I reverted back to showing just the link names.

Reply:
I opened an issue https://github.com/o3de/o3de-extras/issues/512 for discussion of the information for user.